js/src/lib/Details/index.js

```diff
@@ -295,14 +311,53 @@ class Details extends Component {
   }
 
   _openFileBrowser = evt => {
-    this.setState({ isBrowsingFiles: true });
+    // Ignore if is already browsing the files (prevent pushing state to the history repeatedly)
+    if (!this.state.isBrowsingFiles) {
+      this._setFilesSearchParam(true);
+
+      this.setState({ isBrowsingFiles: true });
+    }
     evt.preventDefault();
   };
 
   _closeFileBrowser = evt => {
+    this._setFilesSearchParam(false);
+
     this.setState({ isBrowsingFiles: false });
     evt.preventDefault();
   };
+
+  // Add/remove the 'files' param from the search (push to the history)
+  _setFilesSearchParam = active => {
+    // The strictNullHandling option is used to avoid that the qs includes a '=' at the end of empty params
+    const search = qs.parse(location.search, {
+      ignoreQueryPrefix: true,
+      strictNullHandling: true,
+    });
+
+    if (active) {
+      search.files = null;
+    } else {
+      delete search.files;
+    }
+
+    window.history.pushState(
+      null,
+      null,
+      qs.stringify(search, { addQueryPrefix: true, strictNullHandling: true })
```

Comment:
It's not working properly in production because the URL is diferent.
It's not removing the '?files' param when I click in 'Back to Details'.

I'll fix this

https://deploy-preview-748--yarnpkg.netlify.com/pt-BR/package/qs?files